src/js/pages/jobs/JobRunHistoryTable.js

```diff
@@ -0,0 +1,219 @@
+import classNames from 'classnames';
+import React from 'react';
+
+import DateUtil from '../../utils/DateUtil';
+import ExpandingTable from '../../components/ExpandingTable';
+import FilterHeadline from '../../components/FilterHeadline';
+import JobStates from '../../constants/JobStates';
+import TaskStates from '../../constants/TaskStates';
+
+class JobRunHistoryTable extends React.Component {
+  handleExpansionClick(row) {
+    this.refs.expandingTable.expandRow(row);
+  }
+
+  getColGroup() {
+    return (
+      <colgroup>
+        <col style={{width: '40%'}} />
+        <col style={{width: '20%'}} />
+        <col style={{width: '20%'}} />
+        <col style={{width: '20%'}} />
+      </colgroup>
+    );
+  }
+
+  getColumnHeading(prop, order, sortBy) {
+    let caretClassNames = classNames({
+      'caret': true,
+      'caret--asc': order === 'asc',
+      'caret--desc': order === 'desc',
+      'caret--visible': sortBy.prop === prop
+    });
+
+    let headingStrings = {
+      'jobID': 'Job ID',
+      'status': 'Status',
+      'startedAt': 'Started',
+      'finishedAt': 'Finished'
+    };
+
+    return (
+      <span>
+        {headingStrings[prop]}
+        <span className={caretClassNames}></span>
+      </span>
+    );
+  }
+
+  getColumnClassName(prop, sortBy, row) {
+    return classNames({
+      'highlight': prop === sortBy.prop,
+      'clickable': row == null
+    });
+  }
+
+  getColumns() {
+    return [
+      {
+        className: this.getColumnClassName,
+        render: this.renderJobIDColumn.bind(this),
+        heading: this.getColumnHeading,
+        prop: 'jobID',
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'status',
+        render: this.renderStatusColumn,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'startedAt',
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'finishedAt',
+        sortable: true
+      }
+    ];
+  }
+
+  getData(job) {
```

Comment:
Same questions as in https://github.com/dcos/dcos-ui/pull/397#discussion_r6722183 – do we really need to map the data?